state/web3.go

```diff
@@ -447,7 +447,7 @@ func (w *Web3) updateHolderCandidates(hc HoldersCandidates, ttype TokenType, cur
 		if fromBalance, exists := hc[logData.From]; exists {
 			hc[logData.From] = new(big.Int).Sub(fromBalance, big.NewInt(1))
 		} else {
-			hc[logData.From] = new(big.Int).Neg(big.NewInt(1))
+			hc[logData.From] = big.NewInt(-1)
```

Comment:
The algorithm regarding the partial balances of the token holders of a range of logs must be explained with a comment.